packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/ObjectBoundRealmResults.kt

```diff
@@ -78,5 +78,8 @@ internal class ObjectBoundRealmResults<E : BaseRealmObject>(
  * Binds a flow to an object lifecycle. It allows flows on queries to complete once the object gets
  * deleted. It is used on sub-queries and backlinks.
  */
-internal fun <T> Flow<T>.bind(reference: RealmObjectReference<out BaseRealmObject>): Flow<T> =
-    this.terminateWhen(reference.asFlow()) { it is DeletedObject }
+internal fun <T> Flow<T>.bind(
+    reference: RealmObjectReference<out BaseRealmObject>,
+    keyPaths: List<String>?
```

Comment:
We could use a null default value on keypaths.

Reply:
After reviewing Clauses comments, this was removed altogether.